src/_tutorials/dart-vm/cmdline.md

```diff
@@ -104,62 +106,64 @@ This tutorial goes into detail about this app in the following sections.
-    return <a href="#" class="dart-popover" data-toggle="popover" title="Standard I/O streams" data-html="true" data-trigger="hover focus" data-content="This code reads from the standard input stream and pipes the data to the standard output stream.">stdin.pipe(stdout);</a>
+    <a href="#" class="dart-popover" data-toggle="popover" title="Standard I/O streams" data-html="true" data-trigger="hover focus" data-content="This line reads from the standard input stream and pipes the data to the standard output stream.">stdin.pipe(stdout);</a>
   } else {
-    return Future.forEach(paths, (path) {
+    for (var path in paths) {
       int lineNumber = 1;
-      <a href="#" class="dart-popover" data-toggle="popover" title="Open a file for reading" data-html="true" data-trigger="hover focus" data-content="Use the File class to represent a file on the native file system.">Stream&lt;List&lt;int&gt;&gt; stream = new File(path).openRead();</a>
-      return stream
-          <a href="#" class="dart-popover" data-toggle="popover" title="Data converters" data-html="true" data-trigger="hover focus" data-content="Convert data as it becomes available on a stream.">.transform(UTF8.decoder)</a>
-          .transform(const LineSplitter())
-          <a href="#" class="dart-popover" data-toggle="popover" title="Get data from a stream" data-html="true" data-trigger="hover focus" data-content="Listen to a stream to get data when it becomes available.">.listen((line)</a> {
-            if (showLineNumbers) {
-              stdout.write('${lineNumber++} ');
-            }
-            stdout.writeln(line);
-          }).asFuture().catchError((_) =&gt; _handleError(path));
-    });
+      <a href="#" class="dart-popover" data-toggle="popover" title="Open the stream" data-html="true" data-trigger="hover focus" data-content="Use the File class to represent a file on the native file system.">Stream lines = new File(path)</a>
+          <a href="#" class="dart-popover" data-toggle="popover" title="Open a file for reading" data-html="true" data-trigger="hover focus" data-content="Read the content from the file asynchronously.">.openRead()</a>
+          <a href="#" class="dart-popover" data-toggle="popover" title="Data converters" data-html="true" data-trigger="hover focus" data-content="Convert data as it becomes available on the stream.">.transform(UTF8.decoder)</a>
+          .transform(const LineSplitter());
+      <a href="#" class="dart-popover" data-toggle="popover" title="Try-catch block" data-html="true" data-trigger="hover focus" data-content="Catch errors by placing the asynchronous code inside a try-catch block.">try</a> {
+        <a href="#" class="dart-popover" data-toggle="popover" title="Get data from the stream" data-html="true" data-trigger="hover focus" data-content="Use 'await for' to loop over the values as they become available on the stream.">await for (var line in lines)</a> {
```

Comment:
Perhaps emphasize that this pauses the program's execution. It does nothing while it waits for the next line.